sqlglot/dialects/dialect.py

```diff
@@ -135,6 +144,11 @@ def get_start_end(token_type: TokenType) -> t.Tuple[t.Optional[str], t.Optional[
 
         klass.generator_class.can_identify = klass.can_identify
 
+        if klass.TYPED_DIVISION:
+            klass.parser_class.FACTOR = {**klass.parser_class.FACTOR, TokenType.SLASH: exp.TypedDiv}
+            klass.generator_class.TRANSFORMS[exp.Div] = _cast_div_operands_sql
```

Comment:
let's not do this and just override the base generator directly, reference the constant TYPED_DIVISION there

Reply:
The Generator doesn't have a reference to the Dialect.
In @GeorgeSittas 's PR we added the flag to both the Generator and the Parser. Happed to switch to that, too.

Reply:
i believe theres magic. if you define a constant in generator, it will auto propagate the dialect constants there, so yea let’s add a constant to generator

Reply:
that's right there's some magic here which is implemented here:

```python
        # Pass required dialect properties to the tokenizer, parser and generator classes
        for subclass in (klass.tokenizer_class, klass.parser_class, klass.generator_class):
            for name, value in dialect_properties.items():
                if hasattr(subclass, name):
                    setattr(subclass, name, value)
```

Reply:
this basically allows you to sync generator/parser/tokenizer constants with those defined in the dialect class if the names match